sprint-2/step-2: добавил redux и подключил dnd

srsad opened the pull request.

Commit:
sprint-2/step-2: добавил redux и подключил dnd

kyzinatra commented on src/components/burger-ingredients/index.jsx:
```diff
@@ -37,29 +37,58 @@ const ingredientStatus = ({ title }) => (
   /**
    * Детальное окно ингредиента
    */
   const [showModal, setShowModal] = useState(false)
```
Отлично! Круто, что вы комментруете свой код

kyzinatra commented on src/components/burger-ingredients/burger-ingredients-group/index.jsx:
```diff
@@ -7,24 +8,48 @@ import cls from "./style.module.css"
   return (
     <div id={INGREDIENT_SECTION_IDS[type]} className="pb-10">
-      <h3 className="text text_type_main-medium">{title}</h3>
+      <h3 className="text text_type_main-medium burger-ingredients-group" data-type={type}>{title}</h3>
```
Отлично! Хорошая декомпозиция компонентов

kyzinatra commented on src/components/burger-ingredients/burger-ingredients-group/index.jsx:
```diff
@@ -7,24 +8,48 @@ import cls from "./style.module.css"
+      <h3 className="text text_type_main-medium burger-ingredients-group" data-type={type}>{title}</h3>
 
       <section className={cls.wrapper}>{ingredientsCards}</section>
     </div>
```
Можно лучше: вместо `defaultProps` более современным и быстым аналогом будет использование значений по умолчанию

`{title,type,ingredientsList = [],handleClick = () => {}}`

kyzinatra commented on src/components/burger-constructor/index.jsx:
```diff
@@ -1,25 +1,30 @@
-import { ingredientShape } from "../../types/common"
+  const ingredientsList = useMemo(() => {
+    const result = [...selectIngredients]
+    if (selectBun) {
```
Отлично! Круто, что вы используете `useMemo()`

kyzinatra commented on src/services/actions/order.js:
```diff
@@ -0,0 +1,31 @@
+export const createOrder = (ingredientsIds) => async (dispatch) => {
+  try {
+    const response = await $api.orders.createOrder(ingredientsIds)
+    const result = await checkReponse(response)
```
Можно лучше: вместе с `redux` вы можете использовать `createAsynkThunk` из redux

srsad replied:
а разве я тогда смогу использовать подобные конструкции?

```js
  async function orderHandler() {
    setLoading(true) // локальный стейт который блокирует кнопку оформления, а выносить в стор жирновато
    await dispatch(createOrder(ingredientsIds))
    if (!errorMessage) {
      setShowModal(true)
    }
    setLoading(false)
  }
```

srsad replied:
а разве я тогда смогу использовать подобные конструкции?

```js
  async function orderHandler() {
    setLoading(true) // локальный стейт который блокирует кнопку оформления, а выносить в стор жирновато
    await dispatch(createOrder(ingredientsIds))
    if (!errorMessage) {
      setShowModal(true)
    }
    setLoading(false)
  }
```

Плюс отображение модалки должно происходить только после успешного оформления

kyzinatra commented on src/shared/common.js:
```diff
@@ -1,3 +1,8 @@
+}
+
 export const TABS_TYPES = {
   BUN: "bun",
```
Отлично! Круто, что вы выносите константы

kyzinatra commented on src/components/burger-constructor/burger-constructor-item/index.jsx:
```diff
@@ -0,0 +1,76 @@
+import cls from './style.module.css'
+
+export const BurgerConstructorItem = ({ agregateIngridient, index }) => {
+  const dispatch = useDispatch()
```
Можно лучше: опечатка `Ingridient` -> `Ingredient`

kyzinatra commented on src/components/burger-constructor/burger-constructor-item/index.jsx:
```diff
@@ -0,0 +1,76 @@
+  }
+
+  dragRef(dropRef(itemRef))
+
```
Отлично! Хорошая реализация DnD

kyzinatra commented on src/components/burger-ingredients/burger-ingredients-item/index.jsx:
```diff
@@ -3,14 +3,33 @@ import {
+    >
       <section className={cls.image}>
         <img src={ingredient.image} alt={ingredient.name} />
       </section>
```
Можно лучше: `section` подразумевает какую-то лочичную часть с заголовком и текстом. Тут лучше использовать `figure` и `figcaption`

kyzinatra commented:
Выглядит очень круто! Видно, что проделана большая работа. Отметил хорошую реализацию DnD и отличное разбиение проекта на компоненты. Но все же есть моменты на которые стоит обратить внимание.
Подробнее: 

**Можно лучше:**

-  вместо `defaultProps` более современным и быстрым аналогом будет использование значений по умолчанию

`{title,type,ingredientsList = [],handleClick = () => {}}`
-  вместе с `redux` вы можете использовать `createAsynkThunk` из redux
-  опечатка `Ingridient` -> `Ingredient`
-  `section` подразумевает какую-то лочичную часть с заголовком и текстом. Тут лучше использовать `figure` и `figcaption`


**Что понравилось:**

-  Круто, что вы коментруете свой код
-  Хорошая декомпозиция компонентов
-  Отличное использование `useMemo()`
-  Круто, что вы выносите константы
-  Хорошая реализация DnD


**Обратите внимание, что работа принимается только после исправления всех `«Надо исправить»`.**

Commit:
sprint-2/step-2: попраилв синтаксис пропров по умолчанию
исправил опечатку Ingridient => Ingredient
поправил семантику BurgerIngredientsItem
